xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -395,60 +377,114 @@ function init() {
-    var modal = $(this);
     var button = $(event.relatedTarget);
     var permalinkValue = button.prop('href');
     // Updating the permalink inside modal.
-    modal.find('.form-control').val(permalinkValue);
-    // Go to permalink location.
-    modal.find('input.btn.btn-primary').click( function() {
-      window.location = permalinkValue;
-    });
+    $(this).find('.form-control').val(permalinkValue);
   });
   /**
    * Permalink: Events triggered after the permalink modal is displayed.
    */
   $(document).on('shown.bs.modal', '#permalinkModal', function (event) {
-    var modal = $(this);
     // Autofocus on permalink field.
-    modal.find('.form-control').focus();
+    $(this).find('.form-control').focus();
   })
   /**
-   * Delete: Events triggered after the delete modal is displayed.
+   * Event on permalinkModal button for going to the permalink location.
+   */
+  $(document).on('click','#permalinkModal input.btn-primary',function(){
+    window.location = $('#permalinkModal').find('.form-control').val();
+  });
+});
+
+/**
+ * Delete a comment.
+ */
+require(['jquery', 'xwiki-events-bridge'], function($) {
+  /**
+   * Getting the button that triggers the modal.
    */
   $(document).on('shown.bs.modal' , '#deleteModal', function (event) {
-    var modal = $(this);
-    var button = $(event.relatedTarget);
+    $(this).data ('relatedTarget', $(event.relatedTarget));
+  });
+  /**
+   * Event on deleteModal button.
+   */
+  $(document).on('click','#deleteModal input.btn-danger',function(){
+    var modal = $('#deleteModal');
+    var button = modal.data('relatedTarget');
     /**
      * Ajax request made for deleting the comment.
-     * Delete the HTML element on succes (replace it with a small notification message). Display error message on failure.
+     * Delete the HTML element on succes (replace it with a small notification message).
+     * Display error message on failure.
+     * Disable the delete button before the request is send, so the user cannot resend it in case it takes longer.
      */
-    modal.find('input.btn.btn-danger').click( function(event) {
-      $.ajax({
-        url : button.prop('href'),
-        success : function() {
-          var comment = button.parentsUntil('.reply');
-          // Replace the comment with a "deleted comment" placeholder.
-          comment.replaceWith(createNotification("$services.localization.render('core.viewers.comments.commentDeleted')"));
-          new XWiki.widgets.Notification("$services.localization.render('core.viewers.comments.commentDeleted')", 'done');
-        },
-        error: function () {
-          new XWiki.widgets.Notification("$services.localization.render('core.viewers.comments.delete.failed')", 'error');
+    $.ajax({
+      url : button.prop('href'),
+      beforeSend : function() {
+        button.prop('disabled', true);
+      },
+      success : function() {
+        var comment = button.parentsUntil('.reply');
```

Comment:
Do you need all the ancestors? I think you need only the comment, so why not:

    button.closest('.xwikicomment')